src/main/java/cl/transbank/onepay/model/Transaction.java

```diff
@@ -18,15 +18,50 @@
     private static final String SEND_TRANSACTION = "sendtransaction";
     private static final String COMMIT_TRANSACTION = "gettransactionnumber";
 
+    /**
+     *
+     * @param cart
+     * @return
+     * @throws IOException
+     * @throws SignatureException
+     * @throws TransactionCreateException
+     *
+     * @deprecated use {@link #create(ShoppingCart, Onepay.Channel)} instead
+     */
+    @Deprecated
     public static TransactionCreateResponse create(@NonNull ShoppingCart cart)
             throws IOException, SignatureException, TransactionCreateException {
-        return create(cart, null);
+        return create(cart, Onepay.DEFAULT_CHANNEL);
     }
 
+    public static TransactionCreateResponse create(@NonNull ShoppingCart cart, @NonNull Onepay.Channel channel)
+            throws IOException, SignatureException, TransactionCreateException {
+        return create(cart, channel, null);
+    }
+
+    /**
+     *
+     * @param cart
+     * @param options
+     * @return
+     * @throws IOException
+     * @throws SignatureException
+     * @throws TransactionCreateException
+     *
+     * @deprecated use {@link #create(ShoppingCart, Onepay.Channel, Options)} instead
+     */
+    @Deprecated
     public static TransactionCreateResponse create(@NonNull ShoppingCart cart, Options options)
             throws IOException, SignatureException, TransactionCreateException {
+        return create(cart, Onepay.DEFAULT_CHANNEL, options);
+    }
+
+    public static TransactionCreateResponse create(@NonNull ShoppingCart cart, @NonNull Onepay.Channel channel, Options options)
+            throws IOException, SignatureException, TransactionCreateException {
+        if (channel == Onepay.Channel.APP && Onepay.getAppScheme() == null)
+            throw new TransactionCreateException("You need to set an appScheme if you want to use APP channel");
```

Comment:
Excellent idea to check for this! 

Shouldn't we also do the same for the MOBILE channel and check that there is a callback set different than the default one?

Reply:
Yes we can, changes are ready to review